src/dmd/e2ir.d

```diff
@@ -3933,7 +3933,7 @@ elem* toElem(Expression e, IRState *irs)
             e = ExpressionsToStaticArray(irs, ale.loc, ale.elements, &stmp, 0, ale.basis);
             e = el_combine(e, el_ptr(stmp));
         }
-        else if (ale.elements)
+        else if (ale.elements && dim)
```

Comment:
Since both branches check for `dim`, I would reorganize the code as follows:

```d
if (dim)
{
      if (tb.ty == Tsarray) { ... }
      else if (ale.elements) { ... }
}
else
{
    e = el_long(TYsize_t, 0);
}
```

Reply:
Done